runtime/compiler/optimizer/OSRGuardInsertion.cpp

```diff
@@ -202,6 +203,14 @@ static bool skipOSRGuardInsertion(TR::Compilation* comp)
 
 int32_t TR_OSRGuardInsertion::perform()
    {
+   if (!comp()->getOption(TR_EnableOSR) ||
```

Comment:
should this just be the getOSRMode since getOption(TR_EnableOSR) and getOption(TR_DisableOSR are not inverses?

Reply:
@andrewcraik getOSRMode is not enough, if we're not under FSD, it'll always return `TR::voluntaryOSR`. Changed it to use `supportsInduceOSR`
```
TR::OSRMode
OMR::Compilation::getOSRMode()
   {
   if (self()->getOption(TR_FullSpeedDebug))
      return TR::involuntaryOSR;
   return TR::voluntaryOSR;
   }
```